### PR TITLE
Collections topic

### PR DESCRIPTION
**Pull request by VasilMitovClouway**
The first task from collections topic is ready for review

### CODE REVIEW

File: src/pagebean/PageBean.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  private Integer numberofpages;
+  private Integer key=0;
+
+  public PageBean(Integer pagesize, String content) {
+    this.pagesize = pagesize;
+    this.content = content;
+    this.numberofpages = (content.length() / pagesize) + (content.length() % pagesize);
+  }
+
+  public void createPages() {
```
Comment (mgenov) [inline src/pagebean/PageBean.java]: Pages should be created in demo.

File: src/pagebean/PageBean.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        if (step == content.length()) {
+          break;
+        }
+        currentstring += content.charAt(step);
+        step++;
+      }
+      pages.put(i, currentstring);
+    }
+  }
+  public String next(){
```
Comment (mgenov) [inline src/pagebean/PageBean.java]: Doesn't look right page content to be returned as string.

File: src/pagebean/PageBean.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Created by clouway on 10.05.16.
+ */
+public class PageBean {
+  private Integer pagesize;
+  private String content;
+  private Map<Integer, String> pages = new LinkedHashMap<>();
```
Comment (mgenov) [inline src/pagebean/PageBean.java]: I'm not sure that you need this map here.

Comment (mgenov) [inline src/pagebean/PageBean.java]: For page bean, probably only currentPage and pageSize are required parameters to be kept for the whole page bean.

File: src/pagebean/PageBean.java (hunk 1 above)
Comment (VasilMitovClouway) [inline src/pagebean/PageBean.java]: Should or Shouldn't ? At the moment they are created in the demo. I wasn't sure which is the better choice. What is your advice?

Comment (mgenov) [inline src/pagebean/PageBean.java]: As demo calls createPages doesn't mean that pages are created there. It just delegates it to the PageBean object.

File: src/pagebean/PageBean.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    return "Page1\n"+pages.get(key-1);
+  }
+  public String lastPage(){
+    key=numberofpages-1;
+    return "Page"+(numberofpages)+"\n"+pages.get(numberofpages-1);
+  }
+  public Integer getCurrentPageNumber(){
+    return key;
+  }
+
```
Comment (mgenov) [inline src/pagebean/PageBean.java]: these empty lines could be removed and code to be re-formatted.

**Comment by VasilMitovClouway** [comment]
Changed first task from Collection topic.

File: src/pagebean/PageBean.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.Map;
+
+/**
+ * Created by clouway on 10.05.16.
+ */
+public class PageBean {
+  public Integer currentpage=0;
+  private Integer pagesize;
+  private List<Object> pages;
+  private Integer from=0;
```
Comment (mgenov) [inline src/pagebean/PageBean.java]: from and to could be local variables

File: src/pagebean/PageBean.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  public String lastPage(){
+    currentpage=pages.size()/pagesize;
+    return next();
+  }
+  public Integer getCurrentPageNumber(){
+    return currentpage;
+  }
+
+}
+
```
Comment (mgenov) [inline src/pagebean/PageBean.java]: these empty lines could be removed

File: src/sentenceoperation/SentenceOperation.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+
+/**
+ * @author Vasil Mitov (v.mitov.clouway@gmail.com)
+ */
+public class SentenceOperation {
```
Comment (mgenov) [inline src/sentenceoperation/SentenceOperation.java]: Word counting is not a single operation and name of this class is misleading.

File: src/pagebean/PageBean.java (hunk 5 above)
Comment (VasilMitovClouway) [inline src/pagebean/PageBean.java]: I am using from and to in the boolean methods for hasNext and hasPrevious. Should I try to make it work without them ?

Comment (mgenov) [inline src/pagebean/PageBean.java]: Yes, try without them.

Comment (VasilMitovClouway) [inline src/pagebean/PageBean.java]: Done

File: src/pagebean/PageBean.java (hunk 6 above)
Comment (VasilMitovClouway) [inline src/pagebean/PageBean.java]: Done.

File: src/sentenceoperation/SentenceOperation.java (hunk 7 above)
Comment (VasilMitovClouway) [inline src/sentenceoperation/SentenceOperation.java]: Changed to countWords

Comment (mgenov) [inline src/sentenceoperation/SentenceOperation.java]: countWords is a verb, not noun.

Comment (VasilMitovClouway) [inline src/sentenceoperation/SentenceOperation.java]: wordsCount sound better? im trying to evade the -ing.

Comment (mgenov) [inline src/sentenceoperation/SentenceOperation.java]: Actually Sentence  is more then enough I suppose. 

How you read this code? 
```java
Sentence helloWorld = new Sentence("Hello World"); 
System.out.println(String.format("'%s' has exactly %d words", helloWorld.text(), helloWorld.wordsCount()));
```

Comment (VasilMitovClouway) [inline src/sentenceoperation/SentenceOperation.java]: I don't really understand the question.

Comment (mgenov) [inline src/sentenceoperation/SentenceOperation.java]: How you compare current demo with the provided snippet.

Comment (VasilMitovClouway) [inline src/sentenceoperation/SentenceOperation.java]: Changes made to my demo.

File: src/sentenceoperation/Sentence.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+      }
+      lettercount.put(currentchar, counter);
+    }
+    for (Map.Entry<Character, Integer> entry : lettercount.entrySet()) {
+      if (entry.getValue() > charcount) {
+        charcount = entry.getValue();
+        charecter = String.valueOf(entry.getKey());
+      }
+    }
+    return "Most common char is '" + charecter + "' seen " + charcount + "  times.";
```
Comment (mgenov) [inline src/sentenceoperation/Sentence.java]: You only have to return mostCommon char, not whole message is displayed in the console from the Demo class. Similar things we had discussed tomorrow for the PageBean  class.

Comment (VasilMitovClouway) [inline src/sentenceoperation/Sentence.java]: Changed.

**Comment by VasilMitovClouway** [comment]
@PageBean. I'm working with objects , and I return string representations of them. I overwrote the method to string in my Person class so that I am able to return them in the next() class. My question is should I add a private method display/content/show/currentPageContent/... or of sorts which will return a string representation to the next() method? And if so should I make it so it displays every Object separately, so that in future reading a programmer can change what is displayed before/after/inbetween these objects ?

**Comment by mgenov** [comment]
I don't get your point. Could you give me some snippet.

**Comment by VasilMitovClouway** [comment]
public void next(){
....
display(pages.subList(from,to))
}
private List<String> display(List<Object> pages){
new stringlist=pages.toString;
(displaying the list string logic here)
}

**Comment by mgenov** [comment]
It's not role of PageBean to figure out how results are going to be rendered on the screen. Just subset of rows.

**Comment by VasilMitovClouway** [comment]
So I just make the next() return a List<String> and in it make new lists which are equal to the page list , and return this List<String>?

**Comment by mgenov** [comment]
Yeah, you have to return only the content of the page.

**Comment by VasilMitovClouway** [comment]
Is it better now ?

**Comment by mgenov** [comment]
LGTM !

**Pull request merged**